pkg/database/workouts.go

```diff
@@ -83,26 +83,35 @@ func NewWorkout(u *User, workoutType WorkoutType, notes string, filename string,
 	return &w, nil
 }
 
+func workoutTypeFromGpxTrackType(gpxType string) (workoutType WorkoutType, err error) {
+	switch strings.ToLower(gpxType) {
+	case "running", "run":
+		return WorkoutTypeRunning, nil
+	case "walking", "walk":
+		return WorkoutTypeWalking, nil
+	case "cycling", "cycle":
+		return WorkoutTypeCycling, nil
+	case "snowboarding":
+		return WorkoutTypeSnowboarding, nil
+	case "skiing":
+		return WorkoutTypeSkiing, nil
+	case "swimming":
+		return WorkoutTypeSwimming, nil
+	case "kayaking":
+		return WorkoutTypeKayaking, nil
+	default:
+		return WorkoutTypeAutoDetect, errors.New("unrecognized workout type in gpx")
```

Comment:
I define errors as a var at the top, because you can then use that to check if the error is "that kind of error".

See eg. https://github.com/jovandeginste/workout-tracker/blob/master/pkg/app/app.go#L107 for an example of this pattern.

Reply:
OTOH, since we never do anything with an actual (non-nil) err, you can use the bool return value. The pattern then is to return TypeX, true (success), of TypeAuto, false (failure), and receive it as
```golang
if wType, ok := workoutTypeFromGpxTrackType(...); ok {
 return wType
}
```

Hope this makes sense?

Reply:
That does seem simpler, will update PR